packages/shared/src/components/fields/PasswordField.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { ReactElement, useState } from 'react';
+import { passwordStrength } from 'check-password-strength';
+import { TextField, TextFieldProps } from './TextField';
+import EyeIcon from '../icons/Eye';
+import EyeCancelIcon from '../icons/EyeCancel';
+
+const passwordStrengthStates = {
+  0: {
+    label: 'Risky',
+    className: 'text-theme-status-error',
+    progress: 'bg-theme-status-error w-1/6',
+  },
+  1: {
+    label: 'Risky',
+    className: 'text-theme-status-error',
+    progress: 'bg-theme-status-error w-1/6',
+  },
+  2: {
+    label: `You're almost there`,
+    className: 'text-theme-status-warning',
+    progress: 'bg-theme-status-warning w-1/4',
+  },
+  3: {
+    label: 'Strong as it gets',
+    className: 'text-theme-status-success',
+    progress: 'bg-theme-status-success w-1/2',
+  },
+};
+
+export function PasswordField({
+  type,
+  ...props
+}: TextFieldProps): ReactElement {
+  const [useType, setUseType] = useState(type);
+  const [passwordStrengthLevel, setPasswordStrengthLevel] = useState<number>(0);
+  const [isValid, setIsValid] = useState<boolean>(null);
+  const hasUserAction = isValid !== null;
+  const hint = () => {
```

Comment:
Why do we need it as a function?

Reply:
We don't, changed it to conditions now :)